test/unit/server/bin/plugin/plugin.js

```diff
@@ -0,0 +1,130 @@
+var root = require('requirefrom')('');
+var plugin = root('src/server/bin/plugin/plugin');
+var expect = require('expect.js');
+var sinon = require('sinon');
+var installer = root('src/server/bin/plugin/pluginInstaller');
+var remover = root('src/server/bin/plugin/pluginRemover');
+var settingParser = root('src/server/bin/plugin/settingParser');
+
+describe('kibana cli', function () {
+
+  describe('plugin installer', function () {
+
+    describe('settings.action', function () {
+
+      var program = {
+        command: function () { return program; },
+        description: function () { return program; },
+        option: function () { return program; },
+        action: function (processCommand) {
+          processCommand();
+        }
+      };
+
+      beforeEach(function () {
+        sinon.stub(remover, 'remove');
+        sinon.stub(installer, 'install');
+      });
+
+      afterEach(function () {
+        remover.remove.restore();
+        installer.install.restore();
+        settingParser.parse.restore();
+      });
+
+      it('should call remove if settings.action is "remove"', function () {
+        sinon.stub(settingParser, 'parse', function () {
+          return {
+            action: 'remove'
+          };
+        });
+
+        plugin(program);
```

Comment:
I made this change, however I was not 100% comfortable with it. In this specific scenario, the code you suggested seemed to be more complex, and just as long as the original code. I've also always tried to make my unit tests as simple and straightforward as they can be. Which inherently leads to more verbose code and some copy-pasting, but in theory leads to less bugs in the test code. (However arguments can be made that any time you copy-paste-edit you run the risk of adding easy-to-miss bugs that way too)

The argument against your suggestion is harder to make regarding the changes I checked in to the progressReporter tests. I followed the suggested pattern for this file and it reduced the code dramatically.